### ISSUE TITLE
Avoid unnecessary load of related objects when updating object

### ISSUE THREAD

**Issue by weaselmetal** [katharsis-project/katharsis-core#335]
Hey guys. I realized that when an object is to be updated on the server (PATCH-ed), katharsis loads the existing resource (and its to-one relations) and copies the data from the JSON payload onto the object that was just loaded.
Assuming a to-one relation and hibernate as persistence framework, the related to-one object will be fetched eagerly by hibernate. Thus, it has been loaded anyway and katharsis would not need to load that same object again.
Is there a way to tell katharsis that a related object will always be present when the root object is loaded, and thus does not need to be loaded by katharsis?
Something like `@JsonApiToOne(fetch = false)` to instruct katharsis to not get the object again could probably work. Same goes for `@JsonApiToMany`.
Thanks!

**Comment by masterspambot** [comment]
@chb0github @ieugen What you think about this?

**Comment by chb0github** [comment]
Can you please post a  sample of a 
`GET /yourentity/:id`


And 
`PATCH /yourentity/:id`


So I can see clearly what you want to achieve? I think I understand but want to see

**Comment by keithdmoore** [comment]
I fail to see the true benefits of this.  If the objects are already fetched by Hibernate, is there really much performance benefit?

**Comment by chb0github** [comment]
There could be a definite benefit. That one relationship could join and number of tables. If it's not being modified then there is no reason to do so. And if the only thing it's needed for is to generate a link then that's a lot of overhead. But I want to see his example

**Comment by weaselmetal** [comment]
I don't know what an example should show? Json payload? Java callstack?
Anyway, I realized that a client may have changed the related object to a different object.
Assume object A,B and C, where A has a toOne relation to B. A>B relation was changed on the client to A>C and client PATCH-es A to the server.
Then this new object C would get loaded through the relationship repository and linked.
So object B which was retrieved eagerly by Hibernate when fetching A (which was unnecessary, but not super costly), is going to be replaced by C. So loading C cannot be avoided, which was my original assumption.
